src/test/java/spoon/test/generics/GenericsTest.java

```diff
@@ -695,36 +725,39 @@ public void testCtSuperTypeHierarchy() throws Exception {
 		
 		//type of first parameter of  method WeddingLunch#eatMe
 		CtTypeReference<?> ctWeddingLunch_X = tWeddingLunch_eatMe.getParameters().get(0).getType();
+		// X is the type parameter of WeddingLunch
 		assertEquals("X", ctWeddingLunch_X.getSimpleName());
 		//type of first parameter of method Lunch#eatMe
 		CtTypeReference<?> ctClassLunch_A = ctClassLunch_eatMe.getParameters().get(0).getType();
 		assertEquals("A", ctClassLunch_A.getSimpleName());
 		
 		//are these two types same?
-		CtSuperTypeHierarchy sthOftrWeddingLunch_Mole = new CtSuperTypeHierarchy(trWeddingLunch_Mole);
+		CtSuperTypeHierarchy typingContextOfDisgust = new CtSuperTypeHierarchy(typeReferenceOfDisgust);
 		
-		//adapt X to scope of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>
-		CtTypeReference<?> adapted = sthOftrWeddingLunch_Mole.adaptType(ctWeddingLunch_X);
-		assertEquals("spoon.test.generics.testclasses.Mole", adapted.getQualifiedName());
+		// in disgust, X of WeddingLunch is bound to "Model"
+		assertEquals("spoon.test.generics.testclasses.Mole", typingContextOfDisgust.adaptType(ctWeddingLunch_X).getQualifiedName());
 		//adapt A to scope of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>
-		adapted = sthOftrWeddingLunch_Mole.adaptType(ctClassLunch_A);
-		assertEquals("spoon.test.generics.testclasses.Mole", adapted.getQualifiedName());
-		//adapt A to scope of enclosing class of CelebrationLunch<Integer,Long,Double>.WeddingLunch<Mole>, which is CelebrationLunch<Integer,Long,Double>
-		adapted = sthOftrWeddingLunch_Mole.getEnclosingGenericTypeAdapter().adaptType(ctClassLunch_A);
-		assertEquals("java.lang.Double", adapted.getQualifiedName());
-		
+
+		// in disgust, the A of Lunch is bound to "Mole"
+		assertEquals("spoon.test.generics.testclasses.Mole", typingContextOfDisgust.adaptType(ctClassLunch_A).getQualifiedName());
+
+		// I don't understand the goal and utility of this one
+		assertEquals("java.lang.Double", typingContextOfDisgust.getEnclosingGenericTypeAdapter().adaptType(ctClassLunch_A).getQualifiedName());
```

Comment:
To simplify, can we create instead an adapter of the super class such as  new CtSuperTypeHierarchy(Lunch).adaptType(ctClassLunch_A).getQualifiedName())? 
Can we remove this method?